File: src/main/java/io/funky/fangs/keep_it_personal/command/ClearDeathPreferencesCommand.java
package io.funky.fangs.keep_it_personal.command;

import com.mojang.brigadier.context.CommandContext;
import io.funky.fangs.keep_it_personal.domain.DeathPreferenceContainer;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.network.chat.Component;

public class ClearDeathPreferencesCommand extends KeepItPersonalCommand {
    public static final String NAME = "nothing";
    public static final String PERMISSION = KeepItPersonalCommand.PERMISSION + '.' + NAME;
    private static final String RESPONSE = "Cleared your preferences!";

    @Override
    public Integer run(CommandContext<CommandSourceStack> context, DeathPreferenceContainer container) {
        container.clearDeathPreferences();
        context.getSource().sendSuccess(() -> Component.literal(RESPONSE), false);
        return SINGLE_SUCCESS;
    }
}

File: src/main/kotlin/io/funky/fangs/keep_it_personal/command/ClearDeathPreferencesCommand.kt
package io.funky.fangs.keep_it_personal.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.text.Text

class ClearDeathPreferencesCommand: KeepItPersonalCommand() {
    companion object {
        const val NAME = "nothing"
    }

    override fun run(
        keepCommandState: KeepCommandState,
        context: CommandContext<ServerCommandSource>,
        player: ServerPlayerEntity
    ): Int {
        keepCommandState.clearPlayerPreferences(player.uuid)
        context.source.sendFeedback({ Text.literal("Cleared your preferences!") }, false)
        return Command.SINGLE_SUCCESS
    }
}
